frontend/src/features/admin-form/common/components/CollaboratorModal/CollaboratorListScreen/PermissionDropdown.tsx

```diff
@@ -1,23 +1,32 @@
+import { useMemo } from 'react'
 import { Text } from '@chakra-ui/react'
 
 import Menu from '~components/Menu'
 
-import { DropdownRole } from './AddCollaboratorInput'
+import { DropdownRole } from '../constants'
 
 export interface PermissionDropdownProps {
   value: DropdownRole
   onChange: (role: DropdownRole) => void
   isLoading: boolean
-
+  allowTransferOwnership: boolean
   buttonVariant?: 'outline' | 'clear'
 }
 
 export const PermissionDropdown = ({
   value,
   onChange,
   isLoading,
+  allowTransferOwnership,
   buttonVariant = 'outline',
 }: PermissionDropdownProps): JSX.Element => {
+  const availableRoles = useMemo(() => {
+    return Object.values(DropdownRole).filter((role) => {
+      // Either not owner role, or owner role and allowTransferOwnership is true.
+      return role !== DropdownRole.Owner || allowTransferOwnership
+    })
```

Comment:
/nit Would typescript allow something like this: 
```javascript
const roles = { ...DropdownRole }

if (!allowTransferOwnership) {
    delete roles[DropdownRole.Owner]
}

return Object.values(roles);
```
(I think it makes clearer what this is doing)

Reply:
A little troublesome due to types (destructuring enums result in an object with all enums marked as required, meaning it will take some casting to allow `delete` keyword).

Will keep it like this for now since it's still understandable